### PR TITLE
Send email of Invoices

### PR DESCRIPTION
**Pull request by srivastava-jyoti**
Targets #865

### CODE REVIEW

**Comment by P4NK4J** [comment]
@srivastava-jyoti The PR contains some merge conflicts, please take a look and assign for review once the conflicts are resolved.

File: app/Http/Controllers/EmailInvoiceController.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+use Modules\User\Entities\User;
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Routing\Controller;
+use Modules\Invoice\Entities\Invoice;
+
+class EmailInvoiceController extends Controller
+{
+    protected $name = 'invoice:invoices-list';
+
+    public function handle()
```
Comment (GauravGusain98) [inline app/Http/Controllers/EmailInvoiceController.php:14]: The name of the function is very generic. Can you update it to something more specific?

File: Modules/Invoice/Resources/views/mail/invoices-list.blade.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                <tbody>
+                    @foreach($invoices as $invoice)
+                        <tr style='border-top: 1px solid #000'>
+                            <td style="padding-right: 10px;">{{ $invoice->project->name}}</td>
+                        </tr>
+                    @endforeach   
+                </tbody>
+           </table>
+        </div>
+        <p><a href={{ route('invoice.index') }}> You can see more details here. </a></p>
```
Comment (GauravGusain98) [inline Modules/Invoice/Resources/views/mail/invoices-list.blade.php:20]: Correct indentation.

File: Modules/Invoice/Resources/views/mail/invoices-list.blade.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    
+        Please reach out in case you want to made some changes in this email. 
+    </div>
+
+</div>
```
Comment (GauravGusain98) [inline Modules/Invoice/Resources/views/mail/invoices-list.blade.php:25]: This div is extra

File: app/Http/Controllers/EmailInvoiceController.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        $invoices = Invoice::with(['client', 'project'])->get();
+        $user = new User;
+        $user->email = 'finance@coloredcow.com';
+        Mail::send(
+            'invoice::mail.invoices-list',
+            ['user' => $user, 'invoices' =>$invoices],
+            function ($m) use ($user) {
+                $m->from('finance@coloredcow.com');
+
```
Comment (GauravGusain98) [inline app/Http/Controllers/EmailInvoiceController.php:24]: We usually create a mail file to send a mail. You can take reference form [here](https://laravel.com/docs/8.x/mail#sending-mail)

Comment (GauravGusain98) [reply app/Http/Controllers/EmailInvoiceController.php:24]: Please send the mail using the approach given in the link.

File: app/Http/Controllers/EmailInvoiceController.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    public function handle()
+    {
+        $invoices = Invoice::with(['client', 'project'])->get();
+        $user = new User;
+        $user->email = 'finance@coloredcow.com';
+        Mail::send(
+            'invoice::mail.invoices-list',
+            ['user' => $user, 'invoices' =>$invoices],
+            function ($m) use ($user) {
+                $m->from('finance@coloredcow.com');
```
Comment (GauravGusain98) [inline app/Http/Controllers/EmailInvoiceController.php:23]: This should come from the config variable. Please confirm with Vaibhav, I think we store this email in .env also.  So you need to check the config variable that uses the env variable for this mail.

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check the comments.

**Pull request closed without merging**